include/sdsl/int_vector_buffer.hpp

```diff
@@ -577,7 +577,28 @@ class int_vector_buffer
         {
             return !(*this == it);
         }
-        inline difference_type operator-(iterator const & it)
+
+        bool operator<(iterator const & it) const
+        {
+            return m_idx < it.m_idx;
```

Comment:
What about `m_ivb`?

It only makes sense to ask for relations if both iterators point to the same container, no?
I think, technically, comparing two iterators referring to different containers/ranges is UB anyway.

Reply:
mhm, I could add `assert` statements that check that m_ivb is non-zero and equal?